### ISSUE TITLE
Bug or change in behaviour in ijson from 2.3 to 2.4

### ISSUE THREAD

**Issue by mcarans** [isagalaev/ijson#76]
I have noticed a change in behaviour between 2.3 and 2.4. It can be easily seen with this code:
```
import ijson
from urllib.request import urlopen
for item in ijson.items(urlopen('https://dataviz.vam.wfp.org/api/getfoodprices?ac=196'), 'item'):
    pass
```
2.3 runs through. 2.4 fails with:
json.decoder.JSONDecodeError: Invalid control character at: line 1 column 33 (char 32)

**Comment by rtobar** [comment]
Note: ijson is now maintained in https://github.com/ICRAR/ijson

@mcarans the JSON content is invalid, as it has non-escaped line feeds (\n) and carriage returns (\r) in its strings and thus fails to be parsed. The python backend incorrectly allow this content to parse, but this was fixed in 2.4 (via #52). On the other hand the other backends, even in 2.3, failed. In 2.4 all fail:

```python
import ijson
import importlib

from urllib.request import urlopen

for backend_name in ('yajl2_c', 'yajl2_cffi', 'yajl2', 'python'):
    try:
        backend = importlib.import_module('ijson.backends.%s' % backend_name)
        print('Backend %s' % backend_name)
    except ImportError:
        continue
    try:
        for item in backend.items(urlopen('https://dataviz.vam.wfp.org/api/getfoodprices?ac=196'), 'item'):
            pass
        print('OK')
    except Exception as e:
        print(e)
```

```
# With 2.3
$> python ~/test.py
Backend yajl2_cffi
b'lexical error: invalid character inside string.\n          hilippine Statistics Authority  ","Philippine Statistics Aut\n                     (right here) ------^\n'
Backend yajl2
lexical error: invalid character inside string.
          hilippine Statistics Authority  ","Philippine Statistics Aut
                     (right here) ------^

Backend python
OK

# With 2.4
$> python ~/test.py 
Backend yajl2_c
lexical error: invalid character inside string.
          hilippine Statistics Authority  ","Philippine Statistics Aut
                     (right here) ------^

Backend yajl2_cffi
lexical error: invalid character inside string.
          hilippine Statistics Authority  ","Philippine Statistics Aut
                     (right here) ------^

Backend yajl2
lexical error: invalid character inside string.
          hilippine Statistics Authority  ","Philippine Statistics Aut
                     (right here) ------^

Backend python
Invalid control character at: line 1 column 33 (char 32)
```

Other tools also fail:

```
$> json_pp < ~/download.json 
invalid character encountered while parsing JSON string, at character offset 2553 (before "\r\n","Philippine St...") at /usr/bin/json_pp line 45.
```

**Comment by mcarans** [comment]
Thanks for your help. I will close this.
